packages/payment-processor/src/payment/conversion-erc20.ts

```diff
@@ -26,8 +26,10 @@ export async function approveErc20ForProxyConversionIfNeeded(
   overrides?: ITransactionOverrides,
 ): Promise<ContractTransaction | void> {
   const network =
-    request.extensions[ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY].values.network ||
-    'mainnet';
+    request.extensions[ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY].values.network;
+  if (!network) {
+    throw new Error(`Payment network currency must have a network`);
+  }
```

Comment:
Can't this be handled in the type definition of extensions ? Now that we have a default value, it should be useless to test the presence of this value at higher level methods.